### ISSUE TITLE
ValueConverter when combined with seeding and primary key always recreates data in migrations

### ISSUE THREAD

**Issue by sonofaforester** [dotnet/efcore#13776]
If you apply a ValueConverter, such as the EnumToStringConverter, to a primary key on a table and you have seed data for that table, when you create new migrations, the seed data is always deleted and recreated in each migration. This appears to be related to  #11131

I think the place where it gets out of whack is line 1595 of MigrationsModelDiffer.cs where it is comparing the string to the enum instead of taking the converter into account:

```c#
var modelValuesChanged
   = sourceProperty.ClrType.UnwrapNullableType() == targetProperty.ClrType.UnwrapNullableType()
   && comparer != null
   && !comparer.Equals(sourceValue, targetValue);
```

### Steps to reproduce
A simple csproj file to reproduce is [here](https://kcftech-my.sharepoint.com/:u:/p/medwards/EdHVNsGHbTlNjmlAo9mqvzcBgUr4gCnm3HQBIDG7uWjH3w?e=st58Wf)

Here is my context:

```c#
    public class BloggingContext : DbContext
    {
        public DbSet<Blog> Blogs { get; set; }
        
        protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
        {
            optionsBuilder.UseSqlite("Data Source=blogging.db");
        }

        protected override void OnModelCreating(ModelBuilder modelBuilder)
        {
            modelBuilder.Entity<Blog>(e =>
            {
                e.Property(e1 => e1.BlogId)
                    .HasConversion(new EnumToStringConverter<MyEnum>()).ValueGeneratedNever();
            });

            modelBuilder.Entity<Blog>().HasData(new Blog()
            {
                BlogId = MyEnum.AnEnum,
                Url = "Hello"
            });

            base.OnModelCreating(modelBuilder);
        }
    }

    public class Blog
    {
        [Key]
        public MyEnum BlogId { get; set; }
        public string Url { get; set; }
    }

    public enum MyEnum
    {
        AnEnum,
        AnotherEnum
    }

```

### Further technical details
EF Core version: 2.1.3
Database Provider: Any

**Comment by ChristopherHaws** [comment]
This also happens to me for columns that are not primary keys or enums. It appears to be related to using a `ValueConverter` or `HasConversion` in general. For example, I have a model with the following property: `public TimeZoneInfo TimeZone { get; set; }` which is using the following conversion: `builder.Property(x => x.TimeZone).IsRequired().HasDefaultValueSql("('UTC')").HasConversion(x => x.Id, x => TimeZoneInfo.FindSystemTimeZoneById(x));`.

**Comment by AndriySvyryd** [comment]
@ChristopherHaws What version are you using? It should happen less in 3.1.x

**Comment by ChristopherHaws** [comment]
@AndriySvyryd I am using 3.1.1 of the tools and nuget package.

**Comment by AndriySvyryd** [comment]
Fixed in a503ee42ded7a98ffaea2ba96d138a6c3af9885c
